File: cases/taint/php/loop-carried-negative/loop_carried_negative.php
<?php
function dfb_source(): int { // DFB-SOURCE: loop-carried-input
    return 1;
}

function dfb_sink(int $value): void {} // DFB-SINK: loop-carried-sink

function run(): void {
    $value = dfb_source();
    for ($iteration = 0; $iteration < 3; $iteration++) {
        $value = 0; // DFB-WITNESS: loop-carried-value
    }
    dfb_sink($value);
}

File: cases/taint/kotlin/loop-carried-negative/LoopCarriedNegative.kt
package dataflowbench

object LoopCarriedNegative {
    fun dfb_source(): Int { // DFB-SOURCE: loop-carried-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: loop-carried-sink

    fun run() {
        var value = dfb_source()
        for (iteration in 0 until 3) {
            value = 0 // DFB-WITNESS: loop-carried-value
        }
        dfb_sink(value)
    }
}
